File: test/services/dt/lib/petstore_impl.dart
import 'generated/petstore/api/pet.dart';
import 'generated/petstore/api/petstore/addpet/in.dart' as addpet;
import 'generated/petstore/api/petstore/addpet/out.dart' as addpet;
import 'generated/petstore/api/petstore/getpet/in.dart' as getpet;
import 'generated/petstore/api/petstore/getpet/out.dart' as getpet;
import 'generated/petstore/api/petstore/listpets/in.dart' as listpets;
import 'generated/petstore/api/petstore/listpets/out.dart' as listpets;
import 'generated/petstore/api/petstore/deletepet/in.dart' as deletepet;
import 'generated/petstore/api/petstore/deletepet/out.dart' as deletepet;

class PetStoreImpl {
  final Map<int, Pet> _pets = {};
  int _nextId = 1;

  void reset() {
    _pets.clear();
    _nextId = 1;
  }

  addpet.out addPet(addpet.in_ arg) {
    final petId = _nextId;
    _nextId += 1;
    final pet = Pet(id: petId, name: arg.name, status: arg.status, tag: arg.tag);
    _pets[petId] = pet;
    return addpet.out(pet: pet);
  }

  getpet.out getPet(getpet.in_ arg) {
    final pet = _pets[arg.id];
    if (pet == null) {
      throw StateError('Pet not found: id=${arg.id}');
    }
    return getpet.out(pet: pet);
  }

  listpets.out listPets(listpets.in_ arg) {
    final sorted = _pets.values.toList()..sort((a, b) => a.id.compareTo(b.id));
    return listpets.out(pets: sorted);
  }

  deletepet.out deletePet(deletepet.in_ arg) {
    final existed = _pets.containsKey(arg.id);
    if (existed) {
      _pets.remove(arg.id);
    }
    return deletepet.out(deleted: existed);
  }
}

File: test/services/cs/ServiceTest/PetStoreImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ServiceTest;

public class PetStoreImpl : Petstore.Api.IPetStore
{
    private readonly Dictionary<long, Petstore.Api.Pet> _pets = new();
    private long _nextId = 1;

    public void Reset()
    {
        _pets.Clear();
        _nextId = 1;
    }

    public Petstore.Api.PetStore.AddPet.Out addPet(Petstore.Api.PetStore.AddPet.In arg)
    {
        long id = _nextId++;
        Petstore.Api.Pet pet = new Petstore.Api.Pet(id, arg.Name, arg.Status, arg.Tag);
        _pets[id] = pet;
        return new Petstore.Api.PetStore.AddPet.Out(pet);
    }

    public Petstore.Api.PetStore.GetPet.Out getPet(Petstore.Api.PetStore.GetPet.In arg)
    {
        if (!_pets.TryGetValue(arg.Id, out Petstore.Api.Pet? pet))
        {
            throw new KeyNotFoundException($"Pet not found: id={arg.Id}");
        }
        return new Petstore.Api.PetStore.GetPet.Out(pet);
    }

    public Petstore.Api.PetStore.ListPets.Out listPets(Petstore.Api.PetStore.ListPets.In arg)
    {
        List<Petstore.Api.Pet> sorted = _pets.Values.OrderBy(p => p.Id).ToList();
        return new Petstore.Api.PetStore.ListPets.Out(sorted);
    }

    public Petstore.Api.PetStore.DeletePet.Out deletePet(Petstore.Api.PetStore.DeletePet.In arg)
    {
        bool existed = _pets.Remove(arg.Id);
        return new Petstore.Api.PetStore.DeletePet.Out(existed);
    }
}
